bosh-dev/lib/bosh/dev/tasks/stemcell.rake

```diff
@@ -73,38 +73,99 @@ namespace :stemcell do
 
-      os_image_versions_file = File.expand_path('../../../../../../bosh-stemcell/os_image_versions.json', __FILE__)
-      os_image_versions = JSON.load(File.open(os_image_versions_file))
-      os_image_version = os_image_versions[args.os_image_key]
-      puts "Using OS image #{args.os_image_key}, version #{os_image_version}"
+      puts "Successfully downloaded BOSH micro release to #{release_path}"
+    rescue RuntimeError => e
+      print_help
+      raise e
+    end
+  end
 
-      os_image_uri = URI.join('http://s3.amazonaws.com/', "#{args.os_image_s3_bucket_name}/", args.os_image_key)
-      os_image_uri.query = URI.encode_www_form([['versionId', os_image_version]])
+  desc 'Build a stemcell with a remote pre-built base OS image and bosh micro release'
+  task :build, [:infrastructure_name, :hypervisor_name, :operating_system_name, :operating_system_version, :agent_name, :os_image_s3_bucket_name, :os_image_key] do |_, args|
+    begin
+      require 'bosh/dev/download_adapter'
+      require 'bosh/dev/stemcell_dependency_fetcher'
 
-      Dir.mktmpdir('os-image') do |download_path|
-        os_image_path = File.join(download_path, 'base_os_image.tgz')
-        downloader = Bosh::Dev::DownloadAdapter.new(Logging.logger($stdout))
-        downloader.download(os_image_uri, os_image_path)
+      logger = Logging.logger($stdout)
+      downloader = Bosh::Dev::DownloadAdapter.new(logger)
+      fetcher = Bosh::Dev::StemcellDependencyFetcher.new(downloader, logger)
 
-        Rake::Task['stemcell:build_with_local_os_image'].invoke(args.infrastructure_name, args.hypervisor_name, args.operating_system_name, args.operating_system_version, args.agent_name, os_image_path)
+      mkdir_p('tmp')
+      if 'no' == ENV['BOSH_MICRO_ENABLED']
+        release_path = '/dev/null'
+      else
+        release_path = File.join(Dir.pwd, 'tmp', "bosh-#{PINNED_MICRO_VERSION}.tgz")
+        fetcher.download_bosh_micro_release(
+          bosh_version: PINNED_MICRO_VERSION,
+          output_path: release_path,
+        )
       end
+      os_image_path = File.join(Dir.pwd, 'tmp', 'base_os_image.tgz')
+      fetcher.download_os_image(
+        bucket_name: args.os_image_s3_bucket_name,
+        key:         args.os_image_key,
+        output_path: os_image_path,
+      )
 
+      Rake::Task['stemcell:build_with_local_os_image_with_bosh_release_tarball'].invoke(args.infrastructure_name, args.hypervisor_name, args.operating_system_name, args.operating_system_version, args.agent_name, os_image_path, release_path)
     rescue RuntimeError => e
       print_help
       raise e
     end
   end
 
-  desc 'Build a stemcell using a local OS image and release'
+  desc 'Build a stemcell using a local OS image and bosh micro release'
   task :build_with_local_os_image_with_bosh_release_tarball, [:infrastructure_name, :hypervisor_name, :operating_system_name, :operating_system_version, :agent_name, :os_image_path, :bosh_release_tarball_path, :build_number] do |_, args|
```

Comment:
Seems like we're no longer using `build_number`. Is there a reason we're pulling the value off the environment (on this task) and not receiving it as an argument?